tsup.config.ts

```diff
@@ -0,0 +1,39 @@
+import { relative, resolve as resolveDir } from 'node:path';
+import { defineConfig, type Options } from 'tsup';
+
+export const createTsupConfig = ({
+	globalName,
+	format = ['esm', 'cjs'],
+	target = 'es2021',
+	sourcemap = true,
+	entry = ['src/index.ts'],
+	skipNodeModulesBundle = true,
+	noExternal,
+	esbuildOptions = (options, context) => {
+		if (context.format === 'cjs') {
+			options.banner = {
+				js: '"use strict";',
+			};
+		}
+	},
+}: ConfigOptions = {}) =>
+	defineConfig({
+		clean: true,
+		dts: true,
+		entry,
+		format,
+		minify: false,
```

Comment:
This needs to be configurable. Not to mention that minify was true in a lot of configs.